src/rebar_prv_shell.erl

```diff
@@ -577,6 +581,45 @@ consult_env_config(State, Filename) ->
             rebar_file_utils:consult_config_terms(State, Terms)
     end.
 
+maybe_set_env_vars(State) ->
+    EnvFile =debug_get_value(env_file, rebar_state:get(State, shell, []), undefined,
+                             "Found env_file from config."),
+    {Opts, _} = rebar_state:command_parsed_args(State),
+    EnvFile1 = debug_get_value(env_file, Opts, EnvFile,
+                               "Found env_file from command line option."),
+
+    case maybe_read_file(EnvFile1) of
+        ignore ->
+            ok;
+        {error, _} ->
+            ?WARN("Failed to read file with environment variables: ~p", [EnvFile1]);
+        {ok, Bin} ->
+            Lines = string:split(unicode:characters_to_list(Bin), "\n", all),
+            [handle_env_var_line(Line) || Line <- Lines]
+    end.
+
+handle_env_var_line(Line) ->
+    Trimmed = string:trim(Line, both, [$ ]),
+    %% ignore lines starting with # and
+    %% fail if there are spaces around =
+    case re:run(Trimmed, "^(?<key>[^#][^\s=]*)=(?<value>[^\s]\.*)",
+                [{capture, [key, value], list}]) of
```

Comment:
might want to pass the unicode option to this line so the line content is properly parsed in such cases? Though I guess it blows up on badly formed data like a binary dump

Reply:
Hm. I would think a binary dump wouldn't be used as a env variable, so unicode is probably safe to require.